src/utils.ts

```diff
@@ -256,3 +260,51 @@ export const getExternalProjectNamesFromDbtLoomConfig = (
   }
   return null;
 };
+
+export const isRelationship = (
+  metadata: TestMetadataRelationships | TestMetadataAcceptedValues,
+): metadata is TestMetadataRelationships => {
+  return (metadata as TestMetadataRelationships).field !== undefined;
+};
+
+export const isAcceptedValues = (
+  metadata: TestMetadataRelationships | TestMetadataAcceptedValues,
+): metadata is TestMetadataAcceptedValues => {
+  return (metadata as TestMetadataAcceptedValues).values !== undefined;
+};
+
+export const getColumnTestConfigFromYml = (
+  tests: any[] | undefined,
+  kwargs: TestMetadataAcceptedValues | TestMetadataRelationships,
+  testName: string,
+) => {
+  const existingConfigs = tests?.filter((t: any) => {
+    if (typeof t === "string") {
+      return t === testName;
+    }
+    const [key] = Object.keys(t);
+    return key === testName;
+  });
+
+  return existingConfigs?.find((t: any) => {
+    if (typeof t === "string") {
+      return t === testName;
+    }
+
+    if (isRelationship(kwargs)) {
+      return (
+        kwargs.field === t.relationships.field &&
+        kwargs.to === t.relationships.to
+      );
+    }
+
+    if (isAcceptedValues(kwargs)) {
+      return (
+        kwargs.values?.sort().toString() ===
+        t.accepted_values.values.sort().toString()
+      );
+    }
+
+    return true;
+  });
```

Comment:
this is the actual fix, which finds the config by values/field and to for generic tests